packages/template-retail-react-app/app/components/with-registration/index.jsx

```diff
@@ -13,20 +13,31 @@ import {useLocation} from 'react-router-dom'
 import {useToast} from '@salesforce/retail-react-app/app/hooks/use-toast'
 import {useCurrentCustomer} from '@salesforce/retail-react-app/app/hooks/use-current-customer'
 
-const withRegistration = (Component) => {
+/**
+ * Higher-order component that modifies the given component's `onClick` to show the login form if
+ * the user is not logged in.
+ * @param {React.Component} component Component to wrap.
+ * @param {(location: Location, locale: string) => boolean} [options.isLoginPage] Function that
+ * determines whether the current page is the "login" page, to avoid showing a duplicate modal.
+ * Defaults to all paths ending with "/login".
+ * @returns {React.Component} wrapped component
+ */
+const withRegistration = (
+    Component,
+    {isLoginPage = (loc) => loc.pathname.endsWith('/login')} = {}
+) => {
     const WrappedComponent = ({onClick = noop, ...passThroughProps}) => {
         const {data: customer} = useCurrentCustomer()
         const authModal = useAuthModal()
+        const showToast = useToast()
         const location = useLocation()
         const {formatMessage, locale} = useIntl()
-        const isLoginPage = new RegExp(`^/${locale}/login$`).test(location.pathname)
-        const showToast = useToast()
 
         const handleClick = (e) => {
             e.preventDefault()
             if (!customer.isRegistered) {
                 // Do not show auth modal if users is already on the login page
-                if (isLoginPage) {
+                if (isLoginPage(location, locale)) {
```

Comment:
is local being used?

Reply:
Not here, but it was in the original implementation, so I figured we'd keep it to make customizations easier.

Reply:
There is only one place that defines the isLoginPage which does not take locale. I don't think we need to do it here. 🤔 .

Reply:
Locale was used previously:
https://github.com/SalesforceCommerceCloud/pwa-kit/blob/9baf1c8cf344cdbe5c1a1c9ac6b7eb3976f68d70/packages/template-retail-react-app/app/components/with-registration/index.jsx#L22

Because our default implementation has multiple login pages ([/login](https://pwa-kit.mobify-storefront.com/login), [/en-GB/login](https://pwa-kit.mobify-storefront.com/en-GB/login), [/global/en-GB/login](https://pwa-kit.mobify-storefront.com/global/en-GB/login)), I decided to change the `isLoginPage` logic. But, to make it easier for customers to implement the previous logic (or any locale-dependent custom logic), I think it's reasonable to also provide `locale`.